substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/sampler/SamplerJfrStackTraceSerializer.java

```diff
@@ -167,8 +169,12 @@ private static int visitFrame(JfrNativeEventWriterData data, CodeInfo codeInfo,
         int numStackTraceElements = 0;
         FRAME_INFO_CURSOR.initialize(codeInfo, ip, false);
         while (FRAME_INFO_CURSOR.advance()) {
+            FrameInfoQueryResult frame = FRAME_INFO_CURSOR.get();
+            if (SubstrateJVM.getStackTraceRepo().getTrimInternalStackTraces() &&
```

Comment:
Since it's constant, I prefer moving the condition before the loop, even if the compiler can do that for us ;)

Reply:
I think this condition must be evaluated loop iteration because the outcome may change whenever we advance the `FRAME_INFO_CURSOR` to check the next frame.

Reply:
Hmm I'm not sure we can move the conditional out of the loop since we must recheck it with each new advancement of `FRAME_INFO_CURSOR`.   `SubstrateJVM.getStackTraceRepo().getTrimInternalStackTraces()` is constant but we can't even move that part outside since we still need to execute the loop to potentially serialize frames.

Reply:
Sorry, I was only talking about moving `SubstrateJVM.getStackTraceRepo().getTrimInternalStackTraces()` outside the loop.